### PR TITLE
Detached indicator

### PR DESCRIPTION
**Pull request by AlexTheLion123**
- adds support for detached indicator plots (not overlaid detached indicator, this will be done in a separate PR https://github.com/tradingstrategy-ai/trade-executor/pull/271)
- Also adds various user options to customize subplots. These include:
      1. Subplot color
      2. Horizontal line through subplot
      3. Relative sizing (detached only)
      4. Vertical spacing (detached only)
- Adds example of detached indicators in notebook
    
detached_indicators

### CODE REVIEW

File: tradeexecutor/visual/technical_indicator.py [hunk 1]
```diff
@@ -33,15 +36,73 @@ def overlay_all_technical_indicators(
...
+        cur_row = 1
+    elif volume_bar_mode == VolumeBarMode.separate:
+        cur_row = 2
+    else:
+        raise ValueError("Unknown volume bar mode: %s" % volume_bar_mode)
+    return cur_row
+
+def _add_hline(fig, start_at, end_at, cur_row, plot: Plot):
+    """Add horizontal line to plot if needed."""
+    if line_val := plot.horizontal_line:
```
Comment (miohtama) [inline tradeexecutor/visual/technical_indicator.py:81]: It's a bad practice to do conditional functions.

It is better to check this at the call site and then only call the function if the conditions are sufficient

File: tradeexecutor/visual/technical_indicator.py [hunk 2]
```diff
@@ -33,15 +36,73 @@ def overlay_all_technical_indicators(
...
+                type="line",
+                x0=start_at,
+                y0=line_val,
+                x1=end_at,
+                y1=line_val,
+                line=dict(color=horizontal_line_colour, width=1),
+                row=cur_row, col=1
+            )
+
+def _get_start_and_end(start_at, end_at, plot):
```
Comment (miohtama) [inline tradeexecutor/visual/technical_indicator.py:100]: Need type decorations in the arguments

File: tradeexecutor/visual/technical_indicator.py [hunk 3]
```diff
@@ -33,15 +36,73 @@ def overlay_all_technical_indicators(
...
+    """Get first row for plot based on volume bar mode."""
+    if volume_bar_mode in {VolumeBarMode.hidden, VolumeBarMode.overlay}:
+        cur_row = 1
+    elif volume_bar_mode == VolumeBarMode.separate:
+        cur_row = 2
+    else:
+        raise ValueError("Unknown volume bar mode: %s" % volume_bar_mode)
+    return cur_row
+
+def _add_hline(fig, start_at, end_at, cur_row, plot: Plot):
```
Comment (miohtama) [inline tradeexecutor/visual/technical_indicator.py:79]: Need type decorations in the arguments

**Review by miohtama — CHANGES_REQUESTED** [review]

**Comment by AlexTheLion123** [comment]
Closing and including code inside #271

**Pull request closed without merging**